src/queries/text.ts

```diff
@@ -60,21 +54,18 @@ const getNodeByText = (
   text: TextMatch,
   options: TextMatchOptions = {}
 ) => {
-  try {
-    const { Text } = require('react-native');
-    const isTextComponent = filterNodeByType(node, Text);
-    if (isTextComponent) {
-      const textChildren = getChildrenAsText(node.props.children, Text);
-      if (textChildren) {
-        const textToTest = textChildren.join('');
-        const { exact, normalizer } = options;
-        return matches(text, textToTest, normalizer, exact);
-      }
+  const isTextComponent = filterNodeByType(node, 'Text');
```

Comment:
The trick is to work with host components, instead of composite. Running `instance.findAll()` on a component without component children means that all passed instances to the `ByText` functions will be host components, meaning we won't even search the children.

The test suite passes. This shouldn't be a problem, since we filter by type in the different queries.

Reply:
The reason we picked `const { Text } = require('react-native');` was to make sure we pick the correct component in case the internal displayName changes. I think it used to be "RNText" or "forwardRef(Text)" at some point

Reply:
Another reason, which is not valid anymore, was to support different Text node implementations. So, I'm fine with having this as "Text" as long as we have some kind of guarantee this won't change too often in the framework due to adding refs, memo or whatnot

Reply:
I'd assume RNTL's own test suite would break with any new version of RN which'd change that. But if we want to add it as a runtime check (for instance if someone uses a version that we don't explicitly test), we could still add a check somewhere to make sure that the matching "string name" for `<Text />` is `Text`.